### PR TITLE
Notification- added subscribe/unsubscribe support

### PR DESCRIPTION
**Pull request by kuzmany**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | y
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Now it's not possible identify what push ids are  subscribe/unsubscribe  on OneSignal. 
This PR added small improvements:
- update enabled status for each push ids based on response in notification CampaignSubscriber onCampaignTriggerAction
- extend notification filter in segments  with new subscribed/unsubscribed option

image

#### Steps to test this PR:
1.  Create segment with Notification filter set to Yes
2.  Create segment with Notification filter set to Subscribed
3.  Create segment with Notification filter set to Unsubscribe
4. Go to website from different supported notification browsers (Opera, Firefox, Chrome) and akcept notification from browser notification
5. Make same mautic profile  on each browser. For example use Mautic form with email field and send form. Should same contact now everyhwere
6. Create campaign with segment source from step 1
7. In these step you should have in push_ids table 3 rows for your contact id, each enabled. 
8.  Go to page from chrome and turn off notifications for website
(quick view - http://www.ibtimes.co.uk/how-turn-off-facebook-push-notifications-google-chrome-1536670)
7. Send notification from campaign to contact
9. Contact should send  to firefox, opera, but not to chrome
10. In these step you should have in push_ids table 3 rows for your contact id, but just two enabled
11. Go to segments and see if filters works (should have 1 contact in all notification segments)

### CODE REVIEW

**Comment by kuzmany** [comment]
Label: Ready to test

File: app/bundles/LeadBundle/Model/ListModel.php [hunk 1]
```diff
@@ -602,6 +602,8 @@ public function getChoiceFields()
                     'list' => [
                         0 => $this->translator->trans('mautic.core.form.no'),
```
Comment (alanhartless) [inline app/bundles/LeadBundle/Model/ListModel.php]: This is confusing to me. Yes/No/Subscribed/Unsubscribed. What does yes/no refer to? Maybe it's just a labeling issue.

Comment (kuzmany) [reply app/bundles/LeadBundle/Model/ListModel.php]: Maybe you can help me with labeling
- in past we save mautic notification OneSignal IDs related to user to push_ids table
- times ago @dongilbert  added new column active to this table
- Yes option means all contacts with at least one notification OneSignal ID
- No option means all contacts without at least one notification OneSignal ID
- Subscribed - means all contacts with at least one notification OneSignal IDs, but with flag active  1 (after subscribe, after send notification we know if recepients OneSignal ID is active)
- Unsubscribe - we have notifciation OneSignal ID for contact, but meanwhile contact disable notification

**Comment by kuzmany** [comment]
Label: WIP

File: app/bundles/LeadBundle/Model/ListModel.php (hunk 1 above)
Comment (kuzmany) [reply app/bundles/LeadBundle/Model/ListModel.php]: The reason why we wanna filter Subscribed/Unsubscribed  that now we can send 3000 notifications, but real number receive notifications is less. We need real number of active devices (OneSignal Contacts)

**Comment by kuzmany** [comment]
Label: Pending feedback

**Comment by mleffler** [comment]
Sorry - moving to 2.14, as this is a feature request.

**Pull request closed without merging**